src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -83,6 +86,13 @@ public bool ValidateSuggestedBlock(Block block, out string? errorMessage, bool v
     {
         IReleaseSpec spec = _specProvider.GetSpec(block.Header);
 
+        int encodedSize = _blockDecoder.GetLength(block, RlpBehaviors.None);
+        if (spec.IsEip7934Enabled && encodedSize > spec.Eip7934MaxRlpBlockSize)
```

Comment:
Potentially we could add Block.Size to avoid doing this calculation multiple times, but I expect it is fairly cheap